src/store/__mocks__/load-data.js

```diff
@@ -0,0 +1,40 @@
+import { getUrl } from '../../config';
+import animals from '../../utils/data/animals.mock';
+import demo from '../../utils/data/demo.mock';
+
+/**
+ * Mimic old deprecated API formats which didn't include newer fields
+ * such as pipelines, layers, tags, etc
+ * @param {object} data A dataset file
+ */
+export const mockAPIFeatureSupport = data => {
+  let dataCopy = Object.assign({}, data);
+  if (window.deletePipelines) {
+    delete dataCopy.selected_pipeline;
+    delete dataCopy.pipelines;
+  }
+  return dataCopy;
+};
+
+/**
+ * Mock asynchronously loading/parsing data
+ * @param {string} path JSON file location. Defaults to main data url from config.js
+ * @return {function} A promise that will return when the file is loaded and parsed
+ */
+const loadJsonData = async (path = getUrl('main')) => {
+  await Promise.resolve();
```

Comment:
Might be worth introducing a small timeout to simulate real world use, which should help catch any race conditions.